### PR TITLE
fix https://github.com/mattn/anko/issues/262 : range

### PR DESCRIPTION
**Pull request by Chyroc**

### CODE REVIEW

File: core/core.go [hunk 1]
```diff
@@ -27,23 +27,30 @@ func Import(env *vm.Env) *vm.Env {
...
-			min = args[0]
-			max = args[1]
+		var start, stop int64
+		var step int64 = 1
+
+		switch argsLen := len(args); argsLen {
+		case 0:
+			panic("range expected 1 arguments, got 0")
+		case 1:
+			start = 0
```
Comment (MichaelS11) [inline core/core.go]: No big deal but start is already 0

File: core/core.go [hunk 2]
```diff
@@ -27,23 +27,30 @@ func Import(env *vm.Env) *vm.Env {
...
-			max = args[0] - 1
-		} else {
-			min = args[0]
-			max = args[1]
+		var start, stop int64
+		var step int64 = 1
+
+		switch argsLen := len(args); argsLen {
+		case 0:
+			panic("range expected 1 arguments, got 0")
```
Comment (MichaelS11) [inline core/core.go]: Change to something like: range needs at least 1 argument, got 0

Comment (Chyroc) [reply core/core.go]: remove it

File: core/core.go (hunk 1 above)
Comment (Chyroc) [reply core/core.go]: remove it

File: core/core.go (hunk 2 above)
Comment (Chyroc) [reply core/core.go]: done

File: core/core.go [hunk 3]
```diff
@@ -27,23 +27,29 @@ func Import(env *vm.Env) *vm.Env {
...
-			max = args[0] - 1
-		} else {
-			min = args[0]
-			max = args[1]
+		var start, stop int64
+		var step int64 = 1
+
+		switch argsLen := len(args); argsLen {
+		case 0:
+			panic("range expected at least 1 arguments, got 0")
```
Comment (MichaelS11) [inline core/core.go]: arguments -> argument

**Comment by MichaelS11** [comment]
Could you add some more tests with 0 in them?
For example range(x,0,x)
For example range(x,x,0)

**Comment by MichaelS11** [comment]
For reference, Interesting way JavaScript does it:

https://github.com/lodash/lodash/blob/master/.internal/baseRange.js

File: core/core.go [hunk 4]
```diff
@@ -27,23 +27,32 @@ func Import(env *vm.Env) *vm.Env {
...
+			}
+		default:
+			panic(fmt.Sprintf("range expected at most 3 arguments, got %d", len(args)))
 		}
+
 		arr := []int64{}
-		for i := min; i <= max; i++ {
+		for i := start; (step > 0 && i < stop) || (step < 0 && i > stop); {
 			arr = append(arr, i)
+			i += step
```
Comment (MichaelS11) [inline core/core.go]: Move into for loop?

**Comment by MichaelS11** [comment]
LGTM

**Comment by mattn** [comment]
Thank you

**Pull request merged**